lms/djangoapps/grades/tests/test_tasks.py

```diff
@@ -446,3 +447,26 @@ def test_course_task_args(self, test_batch_size):
             self.assertEqual(batch_size, test_batch_size)
             self.assertEqual(offset, offset_expected)
             offset_expected += test_batch_size
+
+
+@ddt.ddt
+class LockedTaskTest(ModuleStoreTestCase):
+    def setUp(self):
+        super(LockedTaskTest, self).setUp()
+        self.course = CourseFactory.create(
+            org='edx',
+            name='course',
+            run='run',
+        )
+        self.user = UserFactory.create()
+        CourseEnrollment.enroll(self.user, self.course.id)
+
+    @patch('lms.djangoapps.grades.tasks.cache.add')
+    @patch('lms.djangoapps.grades.tasks.cache.delete')
+    @patch('lms.djangoapps.grades.tasks.compute_grades_for_course_v2')
+    @ddt.data(True, False)
+    def test_locked_task(self, cache_open, compute_mock, delete_mock, add_mock):
+        add_mock.return_value = cache_open
+        compute_all_grades_for_course(course_key=unicode(self.course.id))
```

Comment:
Just to make this a little more life-like, can we call `compute_all_grades_for_course.apply_async(kwargs={'course_key': unicode(self.course.id)})`, just to make sure that the celery `task` decorator and our new lock decorator play nicely together?